appveyor.yml

```diff
@@ -1,18 +1,42 @@
-version: 1.0.{build}
-configuration: Debug
 platform: Any CPU
 image: Visual Studio 2017
 install:
 - git submodule update --init --recursive
-# force update ILSpy-tests to the latest ref
-- git submodule update --remote ILSpy-tests
+- ps: |
+    function gitVersion(){
+      $majorVersion = 3;
+      $minorVersion = 0;
+      $revision = 0;
+      $baseCommitHash = "d779383cb85003d6dabeb976f0845631e07bf463";
+      if ($env:APPVEYOR_REPO_BRANCH -ne 'master') {
+        $branch = "-$env:APPVEYOR_REPO_BRANCH";
+      } else {
+        $branch = "";
+      }
+
+      $build = git rev-list --count "$baseCommitHash..HEAD";
+
+      $newVersion="$majorVersion.$minorVersion.$revision.$build";
+      
+      write-host "Update appveyor build version to:$newVersion"
+      $env:ilspy_version_number=$newVersion
+      $env:appveyor_build_version="$newVersion$branch"
+      appveyor UpdateBuild -Version "$newVersion$branch"
+    }
+    gitVersion
 before_build:
 - nuget restore ILSpy.sln
 build_script:
-- msbuild ILSpy.sln /v:minimal /p:ResolveNuGetPackages=false "/logger:%ProgramFiles%\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll"
+- msbuild ILSpy.sln /v:minimal /p:ResolveNuGetPackages=false;Configuration=Debug "/logger:%ProgramFiles%\AppVeyor\BuildAgent\Appveyor.MSBuildLogger.dll"
```

Comment:
:bulb: It would be better to use a matrix for the configuration property.

You can build Debug and Release [like this](https://github.com/Microsoft/perfview/blob/3bc1978007655540240edbfe00a73d558cecbcc3/appveyor.yml#L8-L10), and use `%configuration%` where you currently have the value hard-coded in the build script.